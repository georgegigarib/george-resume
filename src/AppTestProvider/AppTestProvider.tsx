import React from 'react'
import { I18nextProvider } from 'react-i18next'
import i18n from '@/locales/i18n.config'
import store from '@/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'

const AppTestProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/" element={children} />
          </Routes>
        </Router>
      </Provider>
    </I18nextProvider>
  )
}

export default AppTestProvider
