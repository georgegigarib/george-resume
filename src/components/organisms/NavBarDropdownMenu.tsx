import { useState } from 'react';
import { IconButton, Menu, MenuItem } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import ThemeSwitcher from '../molecules/ThemeSwitcher';
import LanguageSwitcher from '../Settings/LanguageSwitcher';

const DropdownMenu = () => {
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);

  const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
  };

  return (
    <div className="absolute right-0">
      <IconButton edge="start" color="inherit" aria-label="menu" onClick={handleMenuOpen}>
        <MenuIcon fontSize='small'/>
      </IconButton>
      <Menu anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleMenuClose}>
        <MenuItem disableRipple className='gap-2'>
          <ThemeSwitcher />
          <LanguageSwitcher  />
        </MenuItem>
      </Menu>
    </div>
  );
};

export default DropdownMenu;
