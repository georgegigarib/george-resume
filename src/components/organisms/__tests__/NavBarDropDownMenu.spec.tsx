import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import NavBarDropdownMenu from '@/components/organisms/NavBarDropdownMenu';
import AppTestProvider from '@/AppTestProvider/AppTestProvider';

describe('DropdownMenu Component', () => {
  const renderComponent = () =>
    render(
      <AppTestProvider>
        <NavBarDropdownMenu />
      </AppTestProvider>,
    );

  it('matches snapshot', () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();

    const menuButton = screen.getByLabelText('menu');
    expect(menuButton).toBeVisible();
  });

  it('opens the menu when the button is clicked', () => {
    const { asFragment } = renderComponent();

    const menuButton = screen.getByLabelText('menu');
    fireEvent.click(menuButton);

    expect(asFragment()).toMatchSnapshot();

    const themeSwitcher = screen.getByTestId('theme-switch-checkbox');
    const languageSwitcher = screen.getByTestId('language-switcher');

    expect(themeSwitcher).toBeVisible();
    expect(languageSwitcher).toBeVisible();
  });

  it('closes the menu when clicked outside', () => {
    const { asFragment } = renderComponent();

    const menuButton = screen.getByLabelText('menu');
    fireEvent.click(menuButton); // Abre el menú

    expect(asFragment()).toMatchSnapshot();

    fireEvent.click(document.body); // Simula clic fuera del menú
    fireEvent.click(document.body); // Simula clic fuera del menú

    const themeSwitcher = screen.queryByTestId('theme-switcher');

    expect(themeSwitcher).not.toBeInTheDocument();
  });
});
