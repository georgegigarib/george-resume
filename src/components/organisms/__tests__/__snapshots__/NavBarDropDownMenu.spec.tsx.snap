// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropdownMenu Component > closes the menu when clicked outside 1`] = `
<DocumentFragment>
  <div
    class="absolute right-0"
  >
    <button
      aria-label="menu"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-efpxe5-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`DropdownMenu Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="absolute right-0"
  >
    <button
      aria-label="menu"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-efpxe5-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`DropdownMenu Component > opens the menu when the button is clicked 1`] = `
<DocumentFragment>
  <div
    class="absolute right-0"
  >
    <button
      aria-label="menu"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-efpxe5-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-18w7uxr-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
